File: go/disgustingEncryption/aes_128_cbc_001.go
package mypck

import (
 "crypto/cipher"
 "crypto/aes"
 "encoding/base64"
 "fmt"
)

type Aes_128_cbc_001 struct {

}

func (a *Aes_128_cbc_001) Show() {
	
	key1 := "12345678"
	str1 := "123"

	enc1 := a.AesEncrypt(str1, key1)
	// 3gVLeGnili1JBTYLHAk8pQ==
	fmt.Println("after encrypt")
	fmt.Println(enc1)

	dec1 := a.AESDecrypt(enc1, key1)
	fmt.Println("after decrypt")
	fmt.Println(dec1)

	key2 := "1234567812345678"
	str2 := "你好abcd1234"
	
	enc2 := a.AesEncrypt(str2, key2)
	// Qkz+MXCIESJZVgHJffouTQ==
	fmt.Println("after encrypt")
	fmt.Println(enc2)

	dec2 := a.AESDecrypt(enc2, key2)
	fmt.Println("after decrypt")
	fmt.Println(dec2)
}



func (a *Aes_128_cbc_001) AesEncrypt(toEncrypt, key string)string{

	keyArray := GetSha256(key)
	toEncryptArray := []byte(toEncrypt)
	iv := []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}

	block, _ := aes.NewCipher(keyArray)
	toEncryptArray = PKCS7Padding(toEncryptArray, block.BlockSize())

	blockMode := cipher.NewCBCEncrypter(block, iv)
		
	resultArray := make([]byte,len(toEncryptArray))
		
	blockMode.CryptBlocks(resultArray, toEncryptArray)
	
	return base64.StdEncoding.EncodeToString(resultArray)
}

func (a *Aes_128_cbc_001) AESDecrypt(toDecrypt, key string)string{
	
	keyArray := GetSha256(key)
	toDecryptArray, _ := base64.StdEncoding.DecodeString(toDecrypt)
	iv:=[]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}

	block,_ := aes.NewCipher(keyArray)

	blockMode := cipher.NewCBCDecrypter(block, iv)

	resultArray := make([]byte,len(toDecryptArray))

	blockMode.CryptBlocks(resultArray, toDecryptArray)

	resultArray = PKCS7UnPadding(resultArray)

	return string(resultArray)
}

File: java/disgustingEncryption/src/main/java/disgustingEncryption/Aes_128_Cbc_001.java
package main.java.disgustingEncryption;

import javax.crypto.Cipher;
import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;
import javax.crypto.spec.IvParameterSpec;
import java.math.BigInteger;
import java.security.SecureRandom;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.nio.charset.StandardCharsets;
import java.util.HashMap;
import java.util.Map;
import java.util.Base64;

public class Aes_128_Cbc_001 {

    private static final String ALGORITHMSTR = "AES/CBC/PKCS5Padding";
    
    public static String aesEncrypt(String content, String encryptKey) throws Exception {

        byte[] keyArray = SHA256Util.getBytes(encryptKey);
        byte[] toEncryptArray = content.getBytes("utf-8");
        byte[] iv = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, };

        SecretKeySpec keySpec = new SecretKeySpec(keyArray, "AES");
        Cipher cipher = Cipher.getInstance(ALGORITHMSTR);        
        cipher.init(Cipher.ENCRYPT_MODE, keySpec, new IvParameterSpec(iv));
        byte[] encData = cipher.doFinal(toEncryptArray);
        return Base64.getEncoder().encodeToString(encData);        
    }

    public static String aesDecrypt(String content, String decryptKey) throws Exception {

        byte[] keyArray = SHA256Util.getBytes(decryptKey);
        byte[] toDecryptArray = Base64.getDecoder().decode(content);
        byte[] iv = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, };

        SecretKeySpec keySpec = new SecretKeySpec(keyArray, "AES");
        Cipher cipher = Cipher.getInstance(ALGORITHMSTR);        
        cipher.init(Cipher.DECRYPT_MODE, keySpec, new IvParameterSpec(iv));
        byte[] resultArray = cipher.doFinal(toDecryptArray);
        return new String(resultArray);
    }

    public static void show(){
        try{
            String key1 = "12345678";
            String str1 = "123";
    
            String enc1 = aesEncrypt(str1, key1);
            // 3gVLeGnili1JBTYLHAk8pQ==
            System.out.println("after encrypt");
            System.out.println(enc1);
    
            String dec1 = aesDecrypt(enc1, key1);
            System.out.println("after decrypt");
            System.out.println(dec1);
    
    
            String key2 = "1234567812345678";
            String str2 = "你好abcd1234";
    
            String enc2 = aesEncrypt(str2, key2);
            // Qkz+MXCIESJZVgHJffouTQ==
            System.out.println("after encrypt");
            System.out.println(enc2);
    
            String dec2 = aesDecrypt(enc2, key2);
            System.out.println("after decrypt");
            System.out.println(dec2);
        } catch (Exception e) {
            e.printStackTrace();
        }        
    }

}